### PR TITLE
Add ability to make anon carts unique

### PR DESCRIPTION
**Pull request by louise-davies**
Added a setting `"anonUserName"` to `topcat.properties`, which is used to check if the cart username is anonymous and to append the session ID if it is

### CODE REVIEW

File: src/main/java/org/icatproject/topcat/web/rest/UserResource.java [hunk 1]
```diff
@@ -62,6 +63,20 @@
 	@PersistenceContext(unitName = "topcat")
 	EntityManager em;
 
+	/**
+	 * Returns the cart userName, which is either the ICAT userName if the user isn't anonUserName,
+	 * or it's the ICAT userName plus the sessionId if it is the anon user name
+	 */
+	private String getCartUserName(String userName, String sessionId) {
+		Properties properties = Properties.getInstance();
+		String anonUserName = properties.getProperty("anonUserName", "");
```
Comment (stuartpullinger) [inline src/main/java/org/icatproject/topcat/web/rest/UserResource.java:72]: Couldn't the properties file access be added to a constructor? It seems that the properties should only be accessed once at instantiation not every time this method is called. This is how the [IdsClient](https://github.com/icatproject/topcat/blob/656e8260216bf1048260ae4ff51f6acfb8af5588/src/main/java/org/icatproject/topcat/IdsClient.java#L38) does it.

**Review by stuartpullinger — CHANGES_REQUESTED** [review]
Looks good. Just one comment.

File: src/main/java/org/icatproject/topcat/web/rest/UserResource.java (hunk 1 above)
Comment (louise-davies) [reply src/main/java/org/icatproject/topcat/web/rest/UserResource.java:72]: True, that would make more sense

**Review by stuartpullinger — APPROVED** [review]
Great. Looks good.

**Pull request merged**